src/twister2/builder/build_manager.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import annotations
+
+import json
+import logging
+import os
+import tempfile
+import time
+from pathlib import Path
+
+import pytest
+from filelock import FileLock
+
+from twister2.builder.builder_abstract import BuildConfig, BuilderAbstract
+
+_TMP_DIR: str = tempfile.gettempdir()
+BUILD_STATUS_FILE_NAME: str = 'twister_builder.json'
+BUILD_LOCK_FILE_PATH: str = os.path.join(_TMP_DIR, 'smart_builder.lock')
+
+logger = logging.getLogger(__name__)
+
+
+class BuildStatus:
+    NOT_DONE = 0
+    IN_PROGRESS = 1
+    DONE = 2
+    FAILED = 3
+
+
+class BuildManager:
+    """
+    Class handles information about already built sources.
+
+    It allows to skip building when it was already built for another test.
+    """
+    _lock: FileLock = FileLock(BUILD_LOCK_FILE_PATH, timeout=1)
+
+    def __init__(self, output_dir: str | Path) -> None:
+        self._status_file: Path = Path(output_dir) / BUILD_STATUS_FILE_NAME
+        self.wait_timeout: int = 10 * 60  # seconds
```

Comment:
Please change name of this variable for more meaningful, like: `wait_build_timeout`.